wpdas/alem
# Feature: complex objects & bug fixes

- Handle complex objects passed as component props: ✅
 - Create new JSX handlers using Babel: ✅
 - Fix issue where components being passed as props is breaking the compiler. It should be compiled, and the component structure must be the same: ✅
 - Fix support for `children` for stateful components: ✅
 - Handle complex conditions to render JSX elements: ✅
 - Comment of type `{/**/}` should be ignored before processing codes: ✅
 - Compiler must support template literals (template strings): ✅
 - Stateful components with the same file name and same component name should be renamed to avoid conflicts: ✅
 - Fix issue where sequential `Social.keys` calls are being merged: ✅
 - Fix function that handle the function's parameters: ✅
 - Fix issue where parenthesis were being removed during the compilation: ✅
 - Fix stateful components props parser: ✅
 - Fix duplicate injections: ✅
 - Use a different name other than `props` for destructuring objects: ✅
 - Stateful components must support rest, e.g.: `<SearchBar {...searchBarProps} />`: ✅
 - Compiler should show an error display if any file use this import statement `import * as foo from "./foo"`: ✅
 - Add support for `async / await` using `promisify` after compiling. A limited functionality was added (experimental): ✅
 - Add `useMemo` types: ✅
 - Add `Big` types: ✅
 - Add layer to check errors without killing Além compiler: ✅
 - Re-test all the features above: ✅
 - Widget transformations need to happen in sub-components as well: ✅
 - Support for arrow functions that return JSX directly without using the return keyword: ✅
 - Add support to use npm modules: ✅ **[experimental]**
 - Verify unused resources and delete them: ✅
